### ISSUE TITLE
Possible memory leak

### ISSUE THREAD

**Issue by davidgs** [kublet/kgfx#2]
I am seeing the Kublet reboot every 20 minutes or so with the following stack trace:

```
[2049920][E][ssl_client.cpp:37] _handle_error(): [start_ssl_client():264]: (-32512) SSL - Memory allocation failed
[2049920][E][WiFiClientSecure.cpp:144] connect(): start_ssl_client: -32512

Malformed array len: cannot fmt
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400d8c3f  PS      : 0x00060130  A0      : 0x800d8d00  A1      : 0x3ffb2100  
A2      : 0x3ffc40e8  A3      : 0x00000004  A4      : 0x000007e0  A5      : 0x3ffc437c  
A6      : 0x00000008  A7      : 0x00000078  A8      : 0x00000000  A9      : 0x00000000  
A10     : 0x0000001c  A11     : 0x00000000  A12     : 0x00000078  A13     : 0x00000000  
A14     : 0x0000001d  A15     : 0x3ffb2114  SAR     : 0x0000001b  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000008  LBEG    : 0x4008b735  LEND    : 0x4008b745  LCOUNT  : 0xffffffff  


Backtrace: 0x400d8c3c:0x3ffb2100 0x400d8cfd:0x3ffb2140 0x400d3739:0x3ffb2170 0x400e0aa9:0x3ffb2290
```

When I run that through the ESP32 Exception Decoder, I get: 

```
PC: 0x400d8c3f: KGFX::drawChart(std::vector   >, int, int, int, int) at .pio/libdeps/esp32dev/KGFX/kgfx.cpp line 123
EXCVADDR: 0x00000008

Decoding stack results
0x400d8c3c: KGFX::drawChart(std::vector   >, int, int, int, int) at .pio/libdeps/esp32dev/KGFX/kgfx.cpp line 122
0x400d8cfd: KGFX::drawChartLarge(std::vector   >, int, int, int) at .pio/libdeps/esp32dev/KGFX/kgfx.cpp line 155
0x400d3739: loop() at src/main.cpp line 89
0x400e0aa9: loopTask(void*) at /Users/davidgs/.platformio/packages/framework-arduinoespressif32/cores/esp32/main.cpp line 50
```

Which looks like a memory leak in the KGFX library. Something isn't being `free()`ed and sooner or later, we run out of memory, and it crashes.

**Comment by audreylim** [comment]
Can you share the running code?

**Comment by audreylim** [comment]
Thanks for the report. Fixed in v0.0.19.

**Comment by davidgs** [comment]
Seems to be fixed! Thanks!
